src/translation.cpp

```diff
@@ -29,23 +31,80 @@ static Translations client_translations;
 Translations *g_client_translations = &client_translations;
 #endif
 
+const std::string_view Translations::getFileLanguage(const std::string &filename)
+{
+	const char *translate_ext[] = {
+		".tr", ".po", ".mo", NULL
+	};
+	auto basename = removeStringEnd(filename, translate_ext);
+	auto pos = basename.rfind('.');
+	if (pos == basename.npos)
+		return "";
+	return basename.substr(pos+1);
+}
 
 void Translations::clear()
 {
 	m_translations.clear();
+	m_plural_translations.clear();
 }
 
 const std::wstring &Translations::getTranslation(
 		const std::wstring &textdomain, const std::wstring &s) const
 {
-	std::wstring key = textdomain + L"|" + s;
+	std::wstring key = textdomain + L"|"; key.append(s);
```

Comment:
```suggestion
	std::wstring key = textdomain + L"|";
	key.append(s);
```

Reply:
I replaced this (and also the following ones) using `string::operator+`. Looks like this is one of the things I took from the original PR without much notice.